auth/TESLA/Sender_Notary.m

```diff
@@ -0,0 +1,111 @@
+classdef Sender_Notary < handle
+    % SENDER_NOTARY Summary of this class goes here
+    %   Detailed explanation goes here
+
+    properties (Access = public)
+        key_chain_length = 100
+        key_disclosure_delay = 5
+        key_chain_function = @HashingWrappers.sha_256
+        key_chain
+        next_key_chain
+    end
+
+    properties (Access = private)
+        sending_interval
+        current_block = 1
+        time_now
+        time_start
+        message_count = 0
+        hmac_num = 0
+        stored_block = cell(1, 5)
+    end
+
+    methods
+
+        function obj = Sender_Notary()
+            obj.key_chain = obj.build_new_keychain;
+            obj.next_key_chain = obj.build_new_keychain;
+        end
+
+        function m = load_new_message(obj, message, time)
+            obj.message_count = obj.message_count + 1;
+            obj.hmac_num = obj.hmac_num + 1;
+
+            if obj.key_chain_length - obj.current_block <= 0
+                obj.switch_hash_path();
+                obj.current_block = 1;
+            end
+
+            % HMAC geration
+            key = obj.key_chain{end - obj.current_block};
+            key = xor(key, time);
+            HMAC = HashingWrappers.hmac_sha_256(message, key);
```

Comment:
Note, the HMAC is truncated to 16-bits (2-bytes), per the standard. This will be caught by a unit test that includes the MT50 encode/decode methods...only 250 bits per message. Right now its 32-bytes.